src/services/services.ts

```diff
@@ -1900,11 +1901,56 @@ namespace ts {
             return Rename.getRenameInfo(program.getTypeChecker(), defaultLibFileName, getCanonicalFileName, getValidSourceFile(fileName), position);
         }
 
+        function getRefactorDiagnostics(fileName: string, range?: TextRange): RefactorDiagnostic[] {
+            synchronizeHostData();
+            const newLineCharacter = host.getNewLine();
+            if (range) {
+                const nonBoundSourceFile = getNonBoundSourceFile(fileName);
+                const context: LightRefactorContext = { newLineCharacter, nonBoundSourceFile };
+                return refactor.getRefactorDiagnosticsForRange(range, context);
+            }
+
+            const boundSourceFile = getValidSourceFile(fileName);
+            const program = getProgram();
+            const context: RefactorContext = { boundSourceFile, newLineCharacter, program, rulesProvider: ruleProvider };
+            const result: RefactorDiagnostic[] = [];
+
+            forEachChild(boundSourceFile, visitor);
+            return result;
+
+            function visitor(node: Node): void {
+                const diags = refactor.getSuggestedRefactorDiagnosticsForNode(node, context);
+                if (diags.length > 0) {
+                    addRange(result, diags);
+                }
+
+                forEachChild(node, visitor);
+            }
+        }
+
+        function getCodeActionsForRefactorAtPosition(

```

Comment:
I don't think so, normally i do that when I have to use the horizontal scroll bar on my laptop if putting all in one line.